File: MailBackend/src/main/java/com/mailapp/mailbackend/dto/ComposeDraftDTO.java
package com.mailapp.mailbackend.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
public class ComposeDraftDTO {
    private Long draftId;
    private List<RecipientComposeDTO> to;
    private List<RecipientComposeDTO> cc;
    private List<RecipientComposeDTO> bcc;
    private String subject;
    private String body;
    private int priority;
    private List<AttachmentDTO> attachments;
}

File: MailFrontend/src/app/models/ComposeDraftDTO.ts
import { Recipient } from "../../features/compose-email/compose-email";
import { AttachmentDTO } from "./DetailedMail";

export interface ComposeDraftDTO {
  draftId: number;
  to: Recipient[];
  cc: Recipient[];
  bcc: Recipient[];
  subject: string;
  body: string;
  priority: number;
  attachments: AttachmentDTO[];
}